quickjs.c

```diff
@@ -573,10 +573,24 @@ typedef enum JSFunctionKindEnum {
 
 #define IC_CACHE_ITEM_CAPACITY 4
 
+typedef int watchpoint_delete_callback(JSRuntime* rt, intptr_t ref, uint8_t offset, JSAtom atom, intptr_t target);
+typedef int watchpoint_free_callback(JSRuntime* rt, intptr_t ref, uint8_t offset, JSAtom atom);
+
+typedef struct JSInlineCacheWatchpoint {
+    watchpoint_delete_callback *delete_callback;
+    watchpoint_free_callback *free_callback;
```

Comment:
These always point to the same two functions so I don't think there's a reason to have this level of indirection. Just call them directly: easier to comprehend, faster to execute, and reduces the size of the struct.